server/src/test/java/org/elasticsearch/common/xcontent/XContentTypeTests.java

```diff
@@ -93,4 +94,65 @@ public void testFromRubbish() throws Exception {
         assertThat(XContentType.fromMediaType("text/plain"), nullValue());
         assertThat(XContentType.fromMediaType("gobbly;goop"), nullValue());
     }
+
+    public void testVersionedMediaType() {
+        String version = String.valueOf(Math.abs(randomByte()));
+        assertThat(XContentType.fromMediaType("application/vnd.elasticsearch+json;compatible-with=" + version),
```

Comment:
can you also add a test that has both compatible-with and charset ?

Reply:
this specific test is only meant to test versioning. I covered multiple parameters support in `MediaTypeParserTests.testJsonWithParameters`

Reply:
thanks... i missed that!